client/dashboard/utils.js

```diff
@@ -49,14 +49,78 @@ export function getProductIdsForCart(
 	profileItems,
 	includeInstalledItems = false,
 	installedPlugins
+) {
+	const productList = getProductList(
+		profileItems,
+		includeInstalledItems,
+		installedPlugins
+	);
+	const productIds = productList.map(
+		( product ) => product.id || product.product
+	);
+	return productIds;
+}
+
+/**
+ * Gets the grouped product names and types for items based on the product types and theme selected in the onboarding profiler.
+ *
+ * @param {Object} profileItems Onboarding profile.
+ * @param {Array} installedPlugins Installed plugins.
+ * @return {Array} Objects with grouped product names and types.
+ */
+export function getGroupedOnboardingProducts( profileItems, installedPlugins ) {
+	const productList = {};
+	productList.products = getProductList(
+		profileItems,
+		true,
+		installedPlugins
+	);
+	productList.remainingProducts = getProductList(
+		profileItems,
+		false,
+		installedPlugins
+	);
+
+	const uniqueItemsList = [
+		...new Set( [
+			...productList.products,
+			...productList.remainingProducts,
+		] ),
+	];
+
+	productList.uniqueItemsList = uniqueItemsList.map( ( product ) => {
+		let cleanedProduct;
+		if ( product.label ) {
+			cleanedProduct = { type: 'extension', name: product.label };
+		} else {
+			cleanedProduct = { type: 'theme', name: product.title };
```

Comment:
I've left other comments about translation - if these `type` values still end up in strings to display (which I don't think they should) they need to be passed through the i18n functions as raw strings here.